text/router-enhancements.md

```diff
@@ -122,7 +124,8 @@ Examples:
 /foo/bar            => 'sample1' (two segments, both static)
 /foo/dynamic        => 'sample2' (two segments, one static, one dynamic)
 /foo/something/else => 'sample3' (two segments, one static, one glob)
-/foo                => 'sample4' (one static segment, but wins over sample 2 because has more handlers)
+/foo                => 'sample2' (one static segment)
+/bar                => 'sample5' (one static segment, but wins over `sample4` because has more handlers)
```

Comment:
This would still be `sample4`.

This definition:
```
this.route('sample4' { path: '/bar/(:optional)' });
```

Should be thought of as:
```
this.route('sample4' { path: '/bar' }) 
this.route('sample4' { path: '/bar/:optional' }) 
```